src/H5TSint.c

```diff
@@ -353,8 +353,13 @@ H5TS__tinfo_create(void)
         HGOTO_DONE(NULL);
 
     /* Reuse an info struct that's on the free list if possible */
-    if (NULL != (tinfo_node = H5TS_tinfo_next_free_s))
+    if (NULL != (tinfo_node = H5TS_tinfo_next_free_s)) {
+        /* Clear the error stack associated with that thread before reuse */
+        if (H5E__clear_stack(&tinfo_node->info.err_stack) < 0)
```

Comment:
It's not about checking the return value, this routine shouldn't be called here.